crates/ruff/src/rules/flake8_async/rules.rs

```diff
@@ -0,0 +1,224 @@
+    &["urllib", "request", "urlopen"],
+    &["httpx", "get"],
+    &["httpx", "post"],
+    &["httpx", "delete"],
+    &["httpx", "patch"],
+    &["httpx", "put"],
+    &["httpx", "head"],
+    &["httpx", "connect"],
+    &["httpx", "options"],
+    &["httpx", "trace"],
+    &["requests", "get"],
+    &["requests", "post"],
+    &["requests", "delete"],
+    &["requests", "patch"],
+    &["requests", "put"],
+    &["requests", "head"],
+    &["requests", "connect"],
+    &["requests", "options"],
+    &["requests", "trace"],
+];
+
+/// ASY100
+pub(crate) fn blocking_http_call(checker: &mut Checker, expr: &Expr) {
+    if in_async_function(&checker.ctx) {
+        if let ExprKind::Call(ast::ExprCall { func, .. }) = &expr.node {
+            if let Some(call_path) = checker.ctx.resolve_call_path(func) {
+                if BLOCKING_HTTP_CALLS.contains(&call_path.as_slice()) {
+                    checker
+                        .diagnostics
+                        .push(Diagnostic::new(BlockingHttpCallInAsyncFunction, func.range));
+                }
+            }
+        }
+    }
+}
+
+/// ## What it does
+/// Checks that async functions do not contain calls to `open`, `time.sleep`,
+/// or `subprocess` methods.
+///
+/// ## Why is this bad?
+/// Blocking an async function via a blocking call will block the entire
+/// event loop, preventing it from executing other tasks while waiting for the
+/// call to complete, negating the benefits of asynchronous programming.
+///
+/// Instead of making a blocking call, use an equivalent asynchronous library
+/// or function.
+///
+/// ## Example
+/// ```python
+/// async def foo():
+///     time.sleep(1000)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// async def foo():
+///     await asyncio.sleep(1000)
+/// ```
+#[violation]
+pub struct OpenSleepOrSubprocessInAsyncFunction;
+
+impl Violation for OpenSleepOrSubprocessInAsyncFunction {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Async functions should not call `open`, `time.sleep`, or `subprocess` methods")
+    }
+}
+
+const OPEN_SLEEP_OR_SUBPROCESS_CALL: &[&[&str]] = &[
+    &["", "open"],
+    &["time", "sleep"],
+    &["subprocess", "run"],
+    &["subprocess", "Popen"],
+    // Deprecated subprocess calls:
+    &["subprocess", "call"],
+    &["subprocess", "check_call"],
+    &["subprocess", "check_output"],
+    &["subprocess", "getoutput"],
+    &["subprocess", "getstatusoutput"],
+    &["os", "wait"],
+    &["os", "wait3"],
+    &["os", "wait4"],
+    &["os", "waitid"],
+    &["os", "waitpid"],
+];
+
+/// ASY101
+pub(crate) fn open_sleep_or_subprocess_call(checker: &mut Checker, expr: &Expr) {
+    if in_async_function(&checker.ctx) {
+        if let ExprKind::Call(ast::ExprCall { func, .. }) = &expr.node {
+            if let Some(call_path) = checker.ctx.resolve_call_path(func) {
+                if OPEN_SLEEP_OR_SUBPROCESS_CALL.contains(&call_path.as_slice()) {
```

Comment:
@qdegraaf - I opted to remove the extra struct to house the candidate members, it didn't feel like it was buying us much, and without it we can just do a simple `.contains` check.